cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -903,11 +927,14 @@ protected Future<Void> podSet() {
                 replicas = kafka.getReplicas();
             }
 
-            StrimziPodSet kafkaPodSet = kafka.generatePodSet(replicas, pfa.isOpenshift(), imagePullPolicy, imagePullSecrets, brokerId -> kafkaPodAnnotations(brokerId, false));
-            return strimziPodSetOperator.reconcile(reconciliation, reconciliation.namespace(), KafkaResources.kafkaStatefulSetName(reconciliation.name()), kafkaPodSet)
-                    .compose(rr -> {
-                        podSetDiff = rr;
-                        return Future.succeededFuture();
+            return secretOperator.getAsync(reconciliation.namespace(), KafkaResources.kafkaSecretName(reconciliation.name()))
```

Comment:
Is the secret not available in the KafkaCluster class somewhere? Do we really need to get it from Kube API? (I'm not saying on 100% it is ... just wondering).

Reply:
I can't find it in the KafkaCluster class

Reply:
The `certificateSecret` is patching the secret. It even already has some logic to decide whether the certificates changed or not. So I guess the secret has to be known there and you can possibly just store the hash stubs there?

Reply:
I've changed this to cache the cert hash in a HashMap that gets populated once when the secret is retrieved.